packages/sdk/client/src/echo/index-query-source-provider.ts

```diff
@@ -68,42 +67,43 @@ export class IndexQuerySource implements QuerySource {
         return [];
       }
 
-      const results: (QueryResult<EchoObject> | undefined)[] = await Promise.all(
-        response.results!.map(async (result) => {
-          const space = this._params.spaceList.get(result.spaceKey);
-          if (!space) {
-            return;
-          }
-
-          await (space as SpaceProxy)._databaseInitialized.wait();
-          const object = await space.db.automerge.loadObjectById(result.id);
-          if (ctx.disposed) {
-            return;
-          }
-
-          if (!object) {
-            return;
-          }
-
-          const core = getAutomergeObjectCore(object);
-          if (!filterMatch(filter, core)) {
-            return;
-          }
-
-          return {
-            id: object.id,
-            spaceKey: core.database!.spaceKey,
-            object,
-            match: { rank: result.rank },
-            resolution: { source: 'index', time: Date.now() - start },
-          };
-        }),
-      );
+      const results: QueryResult<EchoObject>[] = (
+        await Promise.all(
+          response.results!.map(async (result) => {
+            const space = this._params.spaceList.get(result.spaceKey);
+            if (!space) {
+              return;
+            }
+
+            await (space as SpaceProxy)._databaseInitialized.wait();
+            const object = await space.db.automerge.loadObjectById(result.id);
+            if (ctx.disposed) {
+              return;
+            }
+
+            if (!object) {
+              return;
+            }
+
+            const core = getAutomergeObjectCore(object);
+
+            return {
+              id: object.id,
+              spaceKey: core.database!.spaceKey,
+              object,
+              match: { rank: result.rank },
+              resolution: { source: 'index', time: Date.now() - start },
+            };
+          }),
+        )
+      ).filter(Boolean) as QueryResult<EchoObject>[];
```

Comment:
if you filter using a guard like `nonNullable` from `@dxos/util` you shouldn't need to cast